components/ProjectCards.tsx

```diff
@@ -0,0 +1,169 @@
+import { default as projectData } from "../data/project_data";
+import { Box, Badge, Image, Flex, Link, Text } from "@chakra-ui/react";
+import { useEffect, useRef, useState } from "react";
+
+type Projects = Array<{
+  title: string;
+  description: string;
+  access_at?: string | null;
+  project_at?: string | undefined | null;
+  type: string;
+  status: string;
+  categories: Array<string>;
+  contact: string | Array<string>;
+  cfa_stage?: string;
+  date: string;
+  thumb?: string | null;
+  stack?: Array<string>;
+  published?: boolean;
+  screenshot?: string;
+  featured?: boolean;
+  archived?: boolean;
+  skipSize: number;
+}>;
+
+interface Props {
+  currentCategory: Set<string>;
+  currentStatus: Set<string>;
+}
+
+const ProjectCards = ({ currentCategory, currentStatus }: Props) => {
+  const [projects, setProjects] = useState<Projects>([]);
+  const isMounted = useRef(false);
+
+  useEffect(() => {
+    if (!isMounted.current) {
+      setProjects(projectData);
+      isMounted.current = true;
+    }
+  }, []);
+
+  useEffect(() => {
+    if (isMounted.current) {
+      // Handles multiselectors
+      if (currentCategory.size > 0 || currentStatus.size > 0) {
+        const searchedProjects = projectData.filter((p) => {
+          let categories = new Set([...p.categories]);
+          let superSet = new Set([...categories, ...currentCategory]);
+          //   Selecting only for categories
+          if (currentCategory.size > 0 && currentStatus.size === 0) {
+            if (superSet.size < categories.size + currentCategory.size) {
+              return p;
+            }
```

Comment:
This control flow is _pretty_ deep. Four levels of `if` checks can be difficult to reason about. I would suggest thinking about ways to reduce the depth of checks, and possibly eliminating the ref. I'm not sure why we'd need a mutable ref here.